test/plots/segmentPlotTests.ts

```diff
@@ -2,159 +2,168 @@
+        assert.deepEqual(xScale.domain(), [0, 1], "y domain includes only the visible segment (first)");
 
-      let plot = new Plottable.Plots.Segment();
-      plot.x(function(d) { return d.x; }, xScale);
-      plot.y(function(d) { return d.y; }, yScale);
-      plot.y2(function(d) { return d.y2; });
-      plot.addDataset(new Plottable.Dataset(staggeredData));
-      plot.autorangeMode("y");
-      plot.renderTo(svg);
+        yScale.domain([0.5, 1.5]);
+        assert.deepEqual(xScale.domain(), [1, 2], "y domain includes only the visible segment (second)");
+      });
+
+      it("autorangeMode(\"y\")", () => {
+        let staggeredData = [
+          { x: 0, y: 0, y2: 1 },
+          { x: 1, y: 1, y2: 2 }
+        ];
+        yScale.padProportion(0);
 
-      assert.deepEqual(yScale.domain(), [0, 2], "y domain includes both visible segments");
+        let plot = new Plottable.Plots.Segment();
+        plot.x((d) => d.x, xScale);
+        plot.y((d) => d.y, yScale);
+        plot.y2((d) => d.y2);
+        plot.addDataset(new Plottable.Dataset(staggeredData));
+        plot.autorangeMode("y");
 
-      xScale.domain([-0.5, 0.5]);
-      assert.deepEqual(yScale.domain(), [0, 1], "y domain includes only the visible segment (first)");
+        plot.renderTo(svg);
 
-      xScale.domain([0.5, 1.5]);
-      assert.deepEqual(yScale.domain(), [1, 2], "y domain includes only the visible segment (second)");
+        assert.deepEqual(yScale.domain(), [0, 2], "y domain includes both visible segments");
 
-      svg.remove();
+        xScale.domain([-0.5, 0.5]);
+        assert.deepEqual(yScale.domain(), [0, 1], "y domain includes only the visible segment (first)");
+
+        xScale.domain([0.5, 1.5]);
+        assert.deepEqual(yScale.domain(), [1, 2], "y domain includes only the visible segment (second)");
+      });
     });
 
-    describe("entitiesIn() returns segments that intersect with the given constraints", () => {
+    describe("SegmentPlot - entitiesIn()", () => {
       let data = [
         { x: 1, x2: 1, y: 1, y2: 4 },
         { x: 2, x2: 3, y: 4, y2: 3 },
         { x: 4, x2: 5, y: 2, y2: 4 },
         { x: 2, x2: 4, y: 1, y2: 1 }];
 
-      it("retrieves the entities that intersect with the bounding box", () => {
-        let plot = new Plottable.Plots.Segment()
+      let svg: d3.Selection<void>;
+      let xScale: Plottable.Scales.Linear;
+      let yScale: Plottable.Scales.Linear;
+      let plot: Plottable.Plots.Segment<number, number>;
+
+      beforeEach(() => {
+        svg = TestMethods.generateSVG(500, 500);
+        xScale = new Plottable.Scales.Linear();
+        yScale = new Plottable.Scales.Linear();
+        plot = new Plottable.Plots.Segment<number, number>()
           .x((d) => d.x, xScale).x2((d) => d.x2)
           .y((d) => d.y, yScale).y2((d) => d.y2);
         plot.addDataset(new Plottable.Dataset(data)).renderTo(svg);
+      });
+
+      afterEach(() => {
+        svg.remove();
```

Comment:
I was actually thinking about the same thing. What are the advantages of doing that in your opinion?

To me, it seems like an overhead to need to remove the svg all the time. The removal of the SVG has nothing to do with the `segmentPlot` itself. Instead, it is part of the cleaning process. Think of it as you setup the sandbox, you do your testing and you remove the sandbox. The sandbox removal should have nothing to do with the testing itself. Maybe we should test the assertion that the svg can be removed somewhere else and then start using it as an axiom (which we already do).

Can't really see the advantages.